src/operations/Completion.cpp

```diff
@@ -426,28 +330,10 @@ void WorkspaceFolder::suggestImports(const Luau::ModuleName& moduleName, const L
                 }
             }
 
-            if (!isRelative)
-            {
-                // Service will be the first part of the path
-                // If we haven't imported the service already, then we auto-import it
-                auto service = requirePath.substr(0, requirePath.find('/'));
-                if (!contains(importsVisitor.serviceLineMap, service))
-                {
-                    auto lineNumber = importsVisitor.findBestLineForService(service, hotCommentsLineNumber);
-                    bool appendNewline = false;
-                    // If there is no firstRequireLine, then the require that we insert will become the first require,
-                    // so we use `.value_or(lineNumber)` to ensure it equals 0 and a newline is added
-                    if (config.completion.imports.separateGroupsWithLine && importsVisitor.firstRequireLine.value_or(lineNumber) - lineNumber == 0)
-                        appendNewline = true;
-                    textEdits.emplace_back(createServiceTextEdit(service, lineNumber, appendNewline));
-                }
-            }
+            platform->handleRequire(requirePath, lineNumber, isRelative, config, importsVisitor.get(), hotCommentsLineNumber, textEdits);
```

Comment:
nit: `handleRequire` -> `handleRequireAutoImport`?

handleRequire could mean something else, such as actually mapping a `require(...)` to a path in a file resolver.